traitlets/traitlets.py

```diff
@@ -791,6 +792,7 @@ def setup_class(cls, classdict):
         BaseDescriptor in the class dict of the newly created ``cls`` before
         calling their :attr:`class_init` method.
         """
+        cls._descriptors = []
```

Comment:
Does this new state need to go in the `__getstate__`?

Reply:
@minrk if I'm not mistaken you don't have to recreate state for classes  when pickling so `_descriptors` and `_trait_default_generators` shouldn't need to be in `__getstate__`. When the module containing these classes is reloaded then these fields should be repopulated automatically.